src/app/application.cpp

```diff
@@ -645,41 +647,52 @@ int Application::exec(const QStringList &params)
     Net::DownloadManager::initInstance();
     IconProvider::initInstance();
 
-    try
-    {
-        BitTorrent::Session::initInstance();
-        connect(BitTorrent::Session::instance(), &BitTorrent::Session::restored, this, [this]()
-        {
-#ifndef DISABLE_WEBUI
-            m_webui = new WebUI(this);
-#ifdef DISABLE_GUI
-            if (m_webui->isErrored())
-                QCoreApplication::exit(1);
-            connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(1); });
+    BitTorrent::Session::initInstance();
+#ifndef DISABLE_GUI
+    UIThemeManager::initInstance();
 
-            const Preferences *pref = Preferences::instance();
+    m_desktopIntegration = new DesktopIntegration(this);
+    m_desktopIntegration->setToolTip(tr("Loading torrents..."));
+#ifndef Q_OS_MACOS
+    auto *desktopIntegrationMenu = new QMenu;
```

Comment:
`desktopIntegrationMenu` should set `Qt::WA_DeleteOnClose`
and remove `delete m_desktopIntegration->menu();`

Reply:
> `desktopIntegrationMenu` should set `Qt::WA_DeleteOnClose` and remove `delete m_desktopIntegration->menu();`

Then the menu will be deleted after the first use, or am I wrong?

Reply:
I read it wrong, it is good here.